earth_enterprise/src/fusion/autoingest/sysman/khAssetManager.cpp

```diff
@@ -606,7 +606,8 @@ khAssetManager::SubmitTask(const std::string &verref, const TaskDef &taskdef,
 {
   // assert that we're already locked
   assert(!mutex.TryLock());
-
+  static std::mutex theMutex;
```

Comment:
Attempting to run BuildIndex in parallel I was able to catch segfaults stemming from here.  The `OnStateChange` calls wound up eventually getting in here from different threads at the same time during load & think.  The simple check on line 608 passes because one of the threads _is_ holding the mutex.  I got a little heavy handed here.  This could be more granular but not sure if it is worth it.